tools/test/detect_targets_test.py

```diff
@@ -0,0 +1,85 @@
+"""
+mbed SDK
+Copyright (c) 2016 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import unittest
+from collections import namedtuple
+from mock import patch, MagicMock
+from tools.detect_targets import get_interface_version
+from tools.test_api import get_autodetected_MUTS_list
+
+"""
+Tests for detect_targets.py
+"""
+
+class DetectTargetsTests(unittest.TestCase):
+    """
+    Test cases for Detect Target functionality
+    """
+
+    def setUp(self):
+        """
+        Called before each test case
+
+        :return:
+        """
+        # Gather a valid mount point from the host machine
+        muts = get_autodetected_MUTS_list()
```

Comment:
This should be mocked so that we don't depend on this test machine having some connected devices.